### PR TITLE
mockkubeapiserver: set generation on objects

### PR DESCRIPTION
**Pull request by justinsb**
Aiming for a little bit more fidelity, also enables us to test
observedGeneration (in future).

### CODE REVIEW

File: mockkubeapiserver/patchresource.go [hunk 1]
```diff
@@ -124,6 +128,13 @@ func (req *patchResource) Run(ctx context.Context, s *MockKubeAPIServer) error {
 		klog.Infof("skipping write, object not changed")
 		return req.writeResponse(existingObj)
 	} else {
+		bumpGeneration := !reflect.DeepEqual(existingObj.Object["spec"], updated.Object["spec"])
+		if bumpGeneration {
```
Comment (tomasaschan) [inline mockkubeapiserver/patchresource.go:132]: nit: Intent might be clearer expressed here if we move the negation to the conditional:
```suggestion
		specIsSame := reflect.DeepEqual(existingObj.Object["spec"], updated.Object["spec"])
		if !specIsSame {
```

File: mockkubeapiserver/putresource.go [hunk 2]
```diff
@@ -89,6 +89,13 @@ func (req *putResource) Run(ctx context.Context, s *MockKubeAPIServer) error {
 		return req.writeResponse(original)
 	}
 
+	bumpGeneration := !reflect.DeepEqual(original.Object["spec"], updated.Object["spec"])
+	if bumpGeneration {
```
Comment (tomasaschan) [inline mockkubeapiserver/putresource.go:93]: For consistency:
```suggestion
	specIsSame := reflect.DeepEqual(original.Object["spec"], updated.Object["spec"])
	if !specIsSame {
```

File: mockkubeapiserver/patchresource.go (hunk 1 above)
Comment (justinsb) [reply mockkubeapiserver/patchresource.go:132]: Great suggestion, thank you.  Applying!

File: mockkubeapiserver/putresource.go (hunk 2 above)
Comment (justinsb) [reply mockkubeapiserver/putresource.go:93]: Thanks again!

**Pull request merged**